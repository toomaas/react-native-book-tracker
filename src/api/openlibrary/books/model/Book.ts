import {Work} from '../types';

export default class Book {
  title: string;
  coverImageId: number;
  authors: string[];

  constructor(work: Work) {
    const {author_name, cover_i, title} = work;
    this.title = title;
    this.coverImageId = cover_i;
    this.authors = author_name;
  }
}
