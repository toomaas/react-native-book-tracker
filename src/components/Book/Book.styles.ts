import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {margin: 5, alignItems: 'center'},
});

export default styles;
