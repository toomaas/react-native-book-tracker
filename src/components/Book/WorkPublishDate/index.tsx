import WorkPublishDate from './WorkPublishDate';

export default WorkPublishDate;
