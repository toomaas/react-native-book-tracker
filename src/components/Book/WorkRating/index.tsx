import WorkRating from './WorkRating';

export default WorkRating;
