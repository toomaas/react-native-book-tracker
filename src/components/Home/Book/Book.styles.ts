import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {alignItems: 'center'},
  title: {marginTop: 5, fontSize: 16},
});

export default styles;
