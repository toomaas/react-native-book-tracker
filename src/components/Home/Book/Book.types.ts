import Work from '../../../api/openlibrary/works/model/Work';

export type BookProps = {
  work: Work;
};
