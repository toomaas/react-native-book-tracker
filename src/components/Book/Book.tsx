import {useNavigation, useTheme} from '@react-navigation/native';
import React, {useEffect} from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import {HomeScreenPropsNavigationProp} from '../../screens/Home/HomeScreen.types';
import BookCover from '../BookCover';
import {BookProps} from './Book.types';

const Book: React.FunctionComponent<BookProps> = props => {
  const {book} = props;
  const {colors} = useTheme();

  const navigation = useNavigation<HomeScreenPropsNavigationProp>();

  useEffect(() => {}, []);

  return (
    <TouchableOpacity
      onPress={() => {
        navigation.navigate('BookScreen', {book});
      }}>
      <View style={{margin: 5, alignItems: 'center'}}>
        <BookCover
          uri={`https://covers.openlibrary.org/b/id/${book.coverImageId}-S.jpg`}
        />
        <Text style={{color: colors.text}}>{book.title}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default Book;
