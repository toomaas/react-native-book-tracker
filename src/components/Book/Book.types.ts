import Book from '../../api/openlibrary/books/model/Book';

export type BookProps = {
  book: Book;
};
