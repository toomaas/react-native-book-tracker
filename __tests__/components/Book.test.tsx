import React from 'react';
import renderer from 'react-test-renderer';
import Book from '../../src/components/Home/Book';

const mockedDispatch = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: mockedDispatch,
    }),
  };
});

describe('WorkCarousel Component', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Book
          work={{
            authors: ['author1'],
            coverImageId: 10473609,
            description: 'desc',
            editionCount: 25,
            firstPublishYear: 2016,
            key: '/works/OL18020194W',
            title: 'It Ends With Us',
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
