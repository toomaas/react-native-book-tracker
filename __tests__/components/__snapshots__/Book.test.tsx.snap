// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkCarousel Component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <ActivityIndicator />
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "marginTop": 5,
          },
          Object {
            "color": "rgb(28, 28, 30)",
          },
        ]
      }
    >
      It Ends With Us
    </Text>
  </View>
</View>
`;
